src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -2197,6 +2197,73 @@ def test_run_commissioning(self):
             raise signals.TestAbortAll("Failed to commission node(s)")
 
 
+async def run_with_error_check(coroutine_func, *args, exception_type=Exception, assert_func=None, error_msg="Unexpected success", **kwargs):
```

Comment:
```suggestion
async def run_with_error_check(coroutine_func, *args, exception_type=ChipStackError, assert_func=None, error_msg="Unexpected success", **kwargs):
```

Why not default to `ChipStackError` instead of `Exception` since it seems to be used everywhere?